seed/analysis_pipelines/better/client.py

```diff
@@ -146,19 +146,30 @@ def is_ready(res):
             response, errors = res[0], res[1]
             if errors:
                 raise Exception('; '.join(errors))
-            return response['generation_result'] == 'COMPLETE'
 
+            if response['generation_result'] == 'COMPLETE':
+                return True
+            elif response['generation_result'] == 'FAILED':
+                raise Exception(f'BETTER failed to generate the portfolio analysis: {response}')
+            else:
+                return False
+
+        POLLING_TIMEOUT_SECS = 300
         try:
             polling.poll(
                 lambda: self.get_portfolio_analysis(better_portfolio_id, better_analysis_id),
                 check_success=is_ready,
-                timeout=60,
-                step=1
+                timeout=POLLING_TIMEOUT_SECS,
+                step=10,  # wait 10 seconds between polls
```

Comment:
polling every second was a bit aggressive IMO

Reply:
I agree, looking at it now I think BETTER recommends this in their API docs.  `Continue to poll the endpoint (perhaps every 10 seconds) until the generation_result field changes from IN_PROGRESS to another state.`

Reply:
Ooh good point, should probably be checking if the status is COMPLETE or not IN_PROGRESS. Will consider adding that later